### PR TITLE
add delete cover feature to edit community form

### LINKED ISSUE #4385: No option to remove community background image
**Issue Type**
- [X] Feature Idea

Currently there's no way to remove a custom-set background image for communities. It would be really helpful to allow to remove the currently set background image and reset it to the default one.

no-remove-background

### PR DESCRIPTION
**Pull request by ThomasRoest**
**Status**
- [] WIP
- [X] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- api
- hyperion (frontend)

Closes #4385

This is what I've got so far.
covertest pull request

### CODE REVIEW

**Comment by ThomasRoest** [comment]
@brianlovin any comments on this?

**Comment by brianlovin** [comment]
@ThomasRoest sorry for the slow review - just pulled everything and this works great! Just one question below about the code and then we can ship this.

File: src/views/communitySettings/components/editForm.js [hunk 1]
```diff
@@ -301,6 +311,13 @@ class EditForm extends React.Component<Props, State> {
         <SectionTitle>Community Settings</SectionTitle>
         <Form onSubmit={this.save}>
           <ImageInputWrapper>
+            {coverPhoto && !/default_images/.test(coverPhoto) && (
```
Comment (brianlovin) [inline src/views/communitySettings/components/editForm.js:314]: I've never seen this `!/default_images/.test(coverPhoto)` syntax before - could you point me to where I can learn what's going on here?

Comment (ThomasRoest) [reply src/views/communitySettings/components/editForm.js:314]: It's a regex method to check if there's a default image or custom (user uploaded image) and to show or hide the delete (X) button.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test

**Comment by ThomasRoest** [comment]
I've added the same feature to the create community form
image 2019-01-17 at 2 05 09 pm

**Review by mxstbr — APPROVED** [review]
This looks amazing, thank you so much @ThomasRoest!!! :100: :100: :100:

File: src/views/communitySettings/components/editForm.js (hunk 1 above)
Comment (brianlovin) [reply src/views/communitySettings/components/editForm.js:314]: Thank you!

**Pull request merged**